src/components/header/authorizedUser/index.tsx

```diff
@@ -1,12 +1,23 @@
-import { useContext } from 'react'
+import { useContext, useEffect, useState } from 'react'
 import { Link } from 'react-router-dom'
 
+import { IProductInfo } from 'interfaces/IProductInfo'
+import { getGeneralInfo } from 'contracts/getGeneralInfo'
 import { UserContext } from 'contexts/userContext'
 import { logout } from 'contracts/logout'
 import styles from './styles.css'
 
 export const AuthorizedUser = () => {
   const { user, setUser } = useContext(UserContext)
+  const [products, setProducts] = useState<IProductInfo>({} as IProductInfo)
```

Comment:
what does see the user if there is no products?

Reply:
In this case, if there are no products, then the user will see undefined. I'll fix it now